balancer_wrapper.go

```diff
@@ -436,17 +472,37 @@ func (acbw *acBalancerWrapper) RegisterHealthListener(listener func(balancer.Sub
 		return
 	}
 
+	serviceName, registerFn := acbw.healthServiceOpts()
 	acbw.ccb.serializer.TrySchedule(func(ctx context.Context) {
 		if ctx.Err() != nil || acbw.ccb.balancer == nil {
 			return
 		}
 		// Don't send updates if a new listener is registered.
 		acbw.healthMu.Lock()
 		defer acbw.healthMu.Unlock()
-		curHD := acbw.healthData
-		if curHD != hd {
+		if acbw.healthData != hd {
+			return
+		}
+		if registerFn == nil {
+			listener(balancer.SubConnState{ConnectivityState: connectivity.Ready})
```

Comment:
Nit: Do mention here that the raw connectivity state is sent to the health listener if client-side health checking is disabled. I'm sure this is documented somewhere, but I couldn't find any mention of it in this file. So, it might be worth throwing a comment for it here. Thanks.

Reply:
Done.